### PR TITLE
fix warnings from Julia 0.5 

### PR DESCRIPTION
**Pull request by jeff-regier**
Hi @kbarbary  -- This fixes the vast majority of warnings from Julia 0.5, without breaking any backward compatibility, I think.

### CODE REVIEW

File: src/libcfitsio.jl [hunk 1]
```diff
@@ -208,7 +208,7 @@ function fits_create_file(filename::AbstractString)
     ptr = Array(Ptr{Void}, 1)
     status = Cint[0]
     ccall((:ffinit,libcfitsio), Cint, (Ptr{Ptr{Void}},Ptr{UInt8},Ptr{Cint}),
-          ptr, bytestring(filename), status)
+          ptr, string(filename), status)
```
Comment (kbarbary) [inline src/libcfitsio.jl]: Should all these cases of `bytestring(x)` just be replaced by `x`? `string(x::String) === x` on 0.5 and `bytestring(x::ASCIIString) === x` on 0.4. Was this necessary on 0.3? I'm having a hard time remembering...

**Comment by jeff-regier** [comment]
I still test a few warnings when I run the tests.

```
julia> Pkg.test("FITSIO")
INFO: Testing FITSIO
WARNING: Method definition cfitsio_typecode(Type{Int64}) in module Libcfitsio at /home/jeff/.julia/v0.5/FITSIO/src/libcfitsio.jl:164 overwritten at /home/jeff/.julia/v0.5/FITSIO/src/libcfitsio.jl:164.
WARNING: Base.ASCIIString is deprecated, use String instead.
  likely near /home/jeff/.julia/v0.5/FITSIO/src/libcfitsio.jl:356
WARNING: Base.ASCIIString is deprecated, use String instead.
  likely near /home/jeff/.julia/v0.5/FITSIO/src/libcfitsio.jl:356

WARNING: deprecated syntax "[a=>b for (a,b) in c]".
Use "Dict(a=>b for (a,b) in c)" instead.
WARNING: Base.ASCIIString is deprecated, use String instead.
  likely near /home/jeff/.julia/v0.5/FITSIO/test/runtests.jl:118
All tests passed.
INFO: FITSIO tests passed
```

If you know how to fix these, please let me know.

File: src/libcfitsio.jl (hunk 1 above)
Comment (jeff-regier) [inline src/libcfitsio.jl]: No idea...

Comment (kbarbary) [inline src/libcfitsio.jl]: Looking at the old NEWS.md entries, this was necessary on v0.2 but not on v0.3 and later. So I think we can just get rid of these conversions.

File: test/runtests.jl [hunk 2]
```diff
@@ -117,7 +117,7 @@ write(f, indata; hdutype=ASCIITableHDU)
 # expected output type for each input type.
 expected_type = @compat(Dict(Int16=>Int32, Int32=>Int32,
                      Float32=>Float64, Float64=>Float64,
-                     ASCIIString=>ASCIIString))
+                     ASCIIString=>Compat.ASCIIString))
```
Comment (kbarbary) [inline test/runtests.jl]: `Compat.ASCIIString=>Compat.ASCIIString`

File: src/libcfitsio.jl [hunk 3]
```diff
@@ -353,8 +353,8 @@ for (a,T,S) in (("ffukys", :ASCIIString, :(Ptr{UInt8})),
                 ("ffukyl", :Bool,        :Cint),
                 ("ffukyj", :Integer,     :Int64))
     @eval begin
-        function fits_update_key(f::FITSFile, key::ASCIIString, value::$T,
-                                 comment::@compat(Union{ASCIIString, Ptr{Void}})=C_NULL)
+        function fits_update_key(f::FITSFile, key::Compat.ASCIIString, value::$T,
```
Comment (kbarbary) [inline src/libcfitsio.jl]: `$T` is sometimes `ASCIIString`. Change `:ASCIIString` to `:(Compat.ASCIIString)` on L352.

**Comment by kbarbary** [comment]
Did this fix all the warnings?

File: src/header.jl [hunk 4]
```diff
@@ -89,12 +89,12 @@ function fits_try_read_keys{T}(f::FITSFile, ::Type{T}, keys)
     for key in keys
         ccall((:ffgkey, libcfitsio), Cint,
               (Ptr{Void},Ptr{UInt8},Ptr{UInt8},Ptr{UInt8},Ptr{Cint}),
-              f.ptr, bytestring(key), value, C_NULL, status)
+              f.ptr, string(key), value, C_NULL, status)
```
Comment (kbarbary) [inline src/header.jl]: did you miss removing this `string()`?

**Comment by jeff-regier** [comment]
Two warnings left:

```
WARNING: Method definition cfitsio_typecode(Type{Int64}) in module Libcfitsio at /home/jeff/.julia/v0.5/FITSIO/src/libcfitsio.jl:164 overwritten at /home/jeff/.julia/v0.5/FITSIO/src/libcfitsio.jl:164.
```
and 
```
WARNING: deprecated syntax "[a=>b for (a,b) in c]".
Use "Dict(a=>b for (a,b) in c)" instead.
```

**Comment by jeff-regier** [comment]
The second warning is easily fixed by replacing line 114 with
```
        map = Dict(keys[i] => i for i in 1:length(keys))
```
but that breaks compatibility with Julia 0.4. I don't see a macro in Compat.jl that solves the problem

**Comment by kbarbary** [comment]
Could just remove the comprehension: initialize an empty `Dict{String, Int}` and fill it in a loop.

The other warning is because `Clong` is `Int64` on your system. Unrelated to Julia version changes, but maybe there's better method redefinition warnings on 0.5. You could try fixing this with a conditional (only define `cfitsio_typecode(Type{Clong})` if `Clong` is not `Int64`) or it can be done in a separate PR.

**Comment by jeff-regier** [comment]
OK, I think this done now. I'll leave the method definition for another PR.

**Comment by kbarbary** [comment]
Thanks Jeff, looks good.

**Pull request merged**